phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/mt/tenantoperation/TenantOperationFactory.java

```diff
@@ -0,0 +1,501 @@
+                            rs = statement.getResultSet();
+                            boolean isSelectCountStatement = query.getStatement().toUpperCase().trim().contains("COUNT(") ? true : false;
+                            org.apache.hadoop.hbase.util.Pair<Long, Long>
+                                    r = phoenixUtil.getResults(query, rs, queryIteration, isSelectCountStatement, startTime);
+                            resultRowCount = r.getFirst();
+                            queryElapsedTime = r.getSecond();
+                        } else {
+                            conn.commit();
+                        }
+                    } catch (Exception e) {
+                        LOGGER.error("Exception while executing query iteration " + queryIteration, e);
+                    } finally {
+                        try {
+                            if (rs != null) rs.close();
+                            if (statement != null) statement.close();
+                            if (conn != null) conn.close();
+
+                        } catch (Throwable t) {
+                            // swallow;
+                        }
+                    }
+                    return new OperationStats(input, startTime, 0, resultRowCount, queryElapsedTime);
+                }
+            };
+        }
+    }
+
+    class UpsertTenantOperationImpl implements TenantOperationImpl {
+
+        @Override public Function<TenantOperationInfo, OperationStats> getMethod() {
+            return new Function<TenantOperationInfo, OperationStats>() {
+
+                @Nullable @Override public OperationStats apply(@Nullable TenantOperationInfo input) {
+
+                    final int batchSize = loadProfile.getBatchSize();
+                    final boolean useBatchApi = batchSize != 0;
+                    final int rowCount = useBatchApi ? batchSize : 1;
+
+                    final UpsertOperation operation = (UpsertOperation) input.getOperation();
+                    final String tenantGroup = input.getTenantGroupId();
+                    final String opGroup = input.getOperationGroupId();
+                    final String tenantId = input.getTenantId();
+                    final Upsert upsert = operation.getUpsert();
+                    final String tableName = input.getTableName();
+                    final String scenarioName = input.getScenarioName();
+                    final List<Column> columns = upsert.getColumn();
+
+                    final String opName = String.format("%s:%s:%s:%s:%s",
+                            scenarioName, tableName, opGroup, tenantGroup, tenantId);
+
+                    long rowsCreated = 0;
+                    long startTime = 0, duration, totalDuration;
+                    SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+                    try (Connection connection = phoenixUtil.getConnection(tenantId)) {
+                        connection.setAutoCommit(true);
```

Comment:
We call commit() later anyways. Is setting this necessary? In case we run UPSERT SELECTS or DELETES with auto-commit, that will change the execution (client side vs server side) and this might be undesirable.